en/contributing.md

```diff
@@ -28,171 +30,261 @@ and listed here:
+- To understand haproxy, see the [upstream configuration manual][upstream-haproxy-manual].
 
-in the appropriate section. Please make sure you submit a Pull Request with a
-description of the new page and why it is needed!
+...
 
+[dhcp]: installconfig-networking-dhcp.md
+[upstream-haproxy-manual]: http://cbonte.github.io/haproxy-dconv/1.6/configuration.html
+```
 
-## Adding screenshots
+Notes:
 
-When adding screenshots place them in `media`. To reference them in
-your page use the syntax `![description](../media/picture.png)`
+- An internal page is referred to by its source filename.
+- Try to use the same label:destination pair throughout the documentation.
 
 
-# Testing or deploying locally
+### Images
 
-First you need to generate the docs from the Markdown. In the root directory
-first get the dependencies and make the docs:
+Place image files under the `media` directory. In terms of linking, they are
+managed very similarly to hyperlinks. However, they are placed on their own
+line; are preceded by an exclamation point; and both the label and destination
+have a specific naming convention:
 
-```bash
-make sysdeps
-make
+```no-highlight
+![alt-attribute][img__webui_descriptor]
 ```
 
-!!! Note: You only need to `make sysdeps` once, after that you'll have all the
-dependencies you'll need to build the docs going forward.
+The bottom of the file will look like:
 
-The documentation makes use of Javascript for some functionality, so in order
-to test the docs properly you will need to have a web server set up. See
-[Ubuntu and Apache](https://help.ubuntu.com/lts/serverguide/httpd.html). The
-DocumentRoot should be the `htmldocs` directory:
+```no-highlight
+[img__webui_descriptor]: ../media/filename__webui_descriptor.png
+```
 
-```bash
-sudo cp -R htmldocs /var/www/htmldocs
+Explanation:
+
+- `filename`: name of file containing the image (omit the extension '.md')
+- `webui`: version of MAAS corresponding to the image of the web UI
+- `alt-attribute`: text that is shown in place of the image if the latter
+  cannot be displayed for some reason
+
+For example:
+
+```no-highlight
+![enable dhcp][img__2.1_enable-dhcp]
+![enable fire alarm][img__enable-fire-alarm]
+
+...
+
+[img__2.1_enable-dhcp]: ../media/installconfig-networking-dhcp__2.1_enable-dhcp.png
+[img__enable-fire-alarm]: ../media/installconfig-networking-dhcp__enable-fire-alarm.png
 ```
 
-You can then point your web browser at your local machine (127.0.0.1/htmldocs)
-to view the files.
+If the image is not of the MAAS web UI then simply omit the version part (e.g.
+do not include '2.1\_' anywhere in the above).
 
-Alternatively, you can use Python to start a simple HTTP server on the docs
-directory. Navigate to the `/htmldocs` directory of the docs and run the
-following:
+
+## Filenames
+
+The naming of files is based upon its location in the menu (see below). This
+makes it easier for the reader and the writer to build up a mental model of how
+everthing is structured.
```

Comment:
Gah. My syntax highlighting goes bonkers sometimes and makes everything yellow, eclipsing my spellchecker warnings.